StripePaymentSheet/StripePaymentSheet/Source/Internal/API Bindings/Link/PaymentDetails.swift

```diff
@@ -20,17 +20,231 @@ typealias ConsumerSessionWithPaymentDetails = (session: ConsumerSession, payment
  */
 final class ConsumerPaymentDetails: Decodable {
     let stripeID: String
+    let details: Details
+    var isDefault: Bool
 
-    init(stripeID: String) {
+    // TODO(csabol) : Billing address
+
+    init(stripeID: String,
+         details: Details,
+         isDefault: Bool) {
         self.stripeID = stripeID
+        self.details = details
+        self.isDefault = isDefault
     }
 
     private enum CodingKeys: String, CodingKey {
         case stripeID = "id"
+        case isDefault
     }
 
     init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
         self.stripeID = try container.decode(String.self, forKey: .stripeID)
+        // The payment details are included in the dictionary, so we pass the whole dict to Details
+        self.details = try decoder.singleValueContainer().decode(Details.self)
+        self.isDefault = try container.decode(Bool.self, forKey: .isDefault)
+    }
+}
+
+// MARK: - Details
+/// :nodoc:
+extension ConsumerPaymentDetails {
+    enum DetailsType: String, CaseIterable, SafeEnumCodable {
+        case card = "CARD"
+        case bankAccount = "BANK_ACCOUNT"
+        case unparsable = ""
+    }
+
+    enum Details: SafeEnumDecodable {
+        case card(card: Card)
+        case bankAccount(bankAccount: BankAccount)
+        case unparsable
+
+        private enum CodingKeys: String, CodingKey {
+            case type
+            case card = "cardDetails"
+            case bankAccount = "bankAccountDetails"
+        }
+
+        // Our JSON structure doesn't align with Swift's expected structure for enums with associated values, so we do custom decoding.
+        init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            let type = try container.decode(DetailsType.self, forKey: CodingKeys.type)
```

Comment:
This strikes me as odd -- in the event we can't decode this thing, i would expect throw to happen, rather than decoding to .unparsable.

seems like we update DetailsType to:
```
    enum DetailsType: String, CaseIterable, SafeEnumCodable {
        case card = "CARD"
        case bankAccount = "BANK_ACCOUNT"
        case unparsable
    }
```

Then, in the init:
```
            var type: DetailsType
            do {
                type = try container.decode(DetailsType.self, forKey: CodingKeys.type)
            } catch {
                type = .unparsable
            }
```

Reply:
I think the intent here is that we recieve an error upstream if the decoding of `card` or `bankAccount` fails, but if it's an unsupported type (e.g. `bitcoin_wallet` or whatever) then we treat it as unparsable instead of failing decoding entirely.

Reply:
If it comes in as `bitcoin_wallet` wouldn't the `try` throw and parsing would fail?

Reply:
Chatted offline - missed that it was SafeEnumCodable, not Codable